zulipterminal/model.py

```diff
@@ -1224,7 +1225,13 @@ def make_reduced_stream_data(stream: Subscription) -> StreamData:
             # different formats
             subscription["color"] = canonicalize_color(subscription["color"])
 
-            self.stream_dict[subscription["stream_id"]] = subscription
+            stream_id = subscription["stream_id"]
+            if stream_id in self.unsubscribed_stream_dict:
+                del self.unsubscribed_stream_dict[stream_id]
+            elif stream_id in self.never_subscribed_stream_dict:
+                del self.never_subscribed_stream_dict[stream_id]
+            self.stream_dict[stream_id] = subscription
```

Comment:
Minor: since the original line is expanded into a 'synchronize subscriptions' block, it seems clearer to have the stream_id extracted earlier?

That could be achieved by simply separating by a blank line, but the rest of this function could benefit from the use of `stream_id` in any case?